packages/demo-nextjs/src/app/admin/(cms)/system/page.tsx

```diff
@@ -1,22 +1,107 @@
+import { projectId } from "@/utils/mock";
+import clsx from "clsx";
+import React, { FC, useCallback, useEffect, useMemo, useState } from "react";
+import { ProjectSettingsContext } from "./Context";
+import { ContextDataItem } from "./types";
+import { componentsTabs, tabs } from "./utils";
 
 const System: FC = () => {
+  const [data, setData] = useState<ProjectSettings | null>(null);
+  const [isFetching, setIsFetching] = useState(false);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        setIsFetching(true);
+        const response = await fetch(`/api/projectSettings?id=${projectId}`);
+        const { data } = await response.json();
+        setData(data);
+      } catch (e) {
+        console.error(e);
+      } finally {
+        setIsFetching(false);
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  const handleUpdateSettings = useCallback(
+    async (settings: ContextDataItem) => {
+      try {
+        const projectSettings = {
+          ...data,
+          ...settings,
+        } as ProjectSettings;
+
+        setIsFetching(true);
+        const response = await fetch("/api/projectSettings", {
+          method: "PUT",
+          body: JSON.stringify({
+            projectSettings,
+            id: projectId,
+          }),
+        });
+
+        if (!response.ok) {
+          throw new Error("Failed to update settings");
+        }
+
+        setData(projectSettings);
+      } catch (e: unknown) {
+        const message = e instanceof Error ? e.message : "Failed to update settings";
+        alert(message);
+      } finally {
+        setIsFetching(false);
+      }
+    },
+    [data],
+  );
+
+  const contextValue = useMemo(
+    () => ({ data, updateSettings: handleUpdateSettings, isFetching }),
+    [data, handleUpdateSettings, isFetching],
+  );
+
   return (
     <Root>
-      <PageTitle breadcrumbs={Breadcrumbs}>System</PageTitle>
-      <CollectionsListWrapper collection="system" config={config} />
+      <PageTitle breadcrumbs={Breadcrumbs}>Project Settings</PageTitle>
+      <ToolbarWrapper />
+      <Content>
+        <KTCardBody className="py-4 d-flex flex-column gap-8">
+          <KTTabs tabs={tabs} className="nav-line-tabs-2x border-0 fs-4 fw-semibold" />
+          <ProjectSettingsContext.Provider value={contextValue}>
+            <div className="tab-content position-relative">
+              {!data && isFetching ? (
+                <Loading />
+              ) : (
+                componentsTabs.map(({ id, Component }, index) => (
+                  <div
+                    key={id}
+                    id={id}
+                    className={clsx("tab-pane", {
+                      active: index === 0,
+                    })}
+                  >
+                    <Component />
+                  </div>
+                ))
+              )}
+            </div>
+          </ProjectSettingsContext.Provider>
+        </KTCardBody>
+      </Content>
```

Comment:
You can extract all of this into smaller, reusable components. 
For example:
![image](https://github.com/user-attachments/assets/723c5b04-0a18-4f3b-872a-0b51cdf0c038)